plugins/modules/proxmox_cluster.py

```diff
@@ -0,0 +1,302 @@
+        pve_fp:
+          description: Proxmox VE fingerprint.
+          type: str
+          sample: "08:B5:B2:F9:EC:01:0B:D0:..."
+        quorum_votes:
+          description: Quorum votes assigned to the node.
+          type: str
+          sample: "1"
+        ring0_addr:
+          description: Address for ring0.
+          type: str
+          sample: "vmbr0"
+    preferred_node:
+      description: The preferred cluster node.
+      type: str
+      sample: "pve2"
+    totem:
+      description: Totem protocol configuration.
+      type: dict
+      contains:
+        cluster_name:
+          description: Cluster name from totem.
+          type: str
+          sample: "devcluster"
+        config_version:
+          description: Config version.
+          type: str
+          sample: "1"
+        interface:
+          description: Interface configuration.
+          type: dict
+        ip_version:
+          description: IP version.
+          type: str
+          sample: "ipv4-6"
+        link_mode:
+          description: Link mode.
+          type: str
+          sample: "passive"
+        secauth:
+          description: Whether secure authentication is on.
+          type: str
+          sample: "on"
+        version:
+          description: Totem protocol version.
+          type: str
+          sample: "2"
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+try:
+    import urllib3
+    from proxmoxer import ProxmoxAPI
+    __MISSING_MODULE__ = False
+except ImportError:
+    __MISSING_MODULE__ = True
+
+if not __MISSING_MODULE__:
+    urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)
+
+
+def create_cluster(proxmox, module):
+    cluster_name = module.params['cluster_name'] or module.params['api_host']
+    payload = {'clustername': cluster_name}
+    link0 = module.params.get('link0')
+    link1 = module.params.get('link1')
+
+    if link0:
+        payload['link0'] = link0
+    if link1:
+        payload['link1'] = link1
+
+    if module.check_mode:
+        module.exit_json(changed=True, msg="Cluster '{}' would be created (check mode).".format(cluster_name), cluster=cluster_name)
+
+    try:
+        proxmox.cluster.config.post(**payload)
+    except Exception as e:
+        module.fail_json(msg="Error while creating cluster: {}".format(str(e)))
+
+    module.exit_json(changed=True, msg="Cluster '{}' created.".format(cluster_name), cluster=cluster_name)
+
+
+def get_cluster_info(proxmox, module):
+    try:
+        data = proxmox.cluster.config.join.get()
+    except Exception as e:
+        module.fail_json(msg="Error while obtaining cluster join information: {}".format(str(e)))
+
+    module.exit_json(changed=False, ansible_facts={'cluster_join_info': data})
+
+
+def join_cluster(proxmox, module):
+    master_ip = module.params['master_ip']
+    fingerprint = module.params['fingerprint']
+    api_password = module.params['api_password']
+    cluster_name = module.params.get('cluster_name')
```

Comment:
could you use `.ge(...)` on all params ?

Reply:
ack, makes absolutely sense

Reply:
I'd probably use `[]` instead of `get()` since the values are always there (with value `None` if not set by the user and when they don't have a default), and you instantly notice that you made a typo (while `get()` simply returns `None` in that case), or later rename a parameter and forget one instance.

But in the end it's up for discussion to set a standard for the collection on how to do this (always use `get()`, always use `[]`, or let every module do whatever it wants, or even let every line in every module do whatever it wants).